### PR TITLE
[FLOC-1849] Show stderr from admin.runner too.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-1849

This doesn't address the underlying issue (which is we don't interact with vagrant and virtualbox in a way that cleans up old images properly). But it does address the regression in #1294, where we stopped showing stderr from vagrant.

(See 
http://build.clusterhq.com/builders/flocker%2Facceptance%2Fvagrant%2Ffedora-20/builds/2059
where it demonstrates the output).

### CODE REVIEW

File: admin/runner.py [hunk 1]
```diff
@@ -27,21 +48,21 @@ class CommandProtocol(LineOnlyReceiver, object):
...
+        self._fds = defaultdict(lambda: _LineParser(output=self.output))
 
-    def connectionMade(self):
-        self.transport.disconnecting = False
+    def childDataReceived(self, childFD, data):
+        self._fds[childFD].dataReceived(data)
 
-    def connectionLost(self, reason):
-        if reason.check(ConnectionDone):
+    def processEnded(self, reason):
```
Comment (tomprince) [inline admin/runner.py]: This should probably actually be `processExited`.

File: admin/test/test_runner.py [hunk 2]
```diff
@@ -105,6 +105,20 @@ def test_unregisters_killer_failure(self):
...
+        the reason.
+        """
+
+        reactor = ProcessCoreReactor()
+        d = run(reactor, ['command', 'and', 'args'])
+        [process] = reactor.processes
+
+        expected_failure = Failure(ProcessDone(0))
+        process.processProtocol.processEnded(expected_failure)
+        self.successResultOf(d),
```
Comment (adamtheturtle) [inline admin/test/test_runner.py]: This should be compared to the reason.

File: admin/runner.py [hunk 3]
```diff
@@ -27,21 +48,21 @@ class CommandProtocol(LineOnlyReceiver, object):
         If the command finished successfully, will fire with ``None``.
         Otherwise, errbacks with the reason.
     :ivar file-like output: For logging.
+
+    :ivar defaultdict _fds: Mapping from fds to `_LineParsers`.
```
Comment (adamtheturtle) [inline admin/runner.py]: :question: It seems to be more standard to write "file descriptors" - at least it'd be easier for me to read.

File: admin/test/test_runner.py (hunk 2 above)
Comment (adamtheturtle) [inline admin/test/test_runner.py]: Actually, the reason for success? That means the docstring should be fixed and this comma should be removed.

File: admin/runner.py (hunk 1 above)
Comment (adamtheturtle) [inline admin/runner.py]: Whichever it ends up as can you comment please - I have read the definitions of both and am not clear on why it would be either.

**Comment by adamtheturtle** [comment]
This appears to work and doesn't seem to break anything. I've looked into some of the Twisted things used but I don't completely understand it. I've asked for some comments but I don't think that it is top priority that this is the shiniest code so merge when you and Buildbot are happy, or resubmit if there are any major changes.

**Pull request merged**